src/test/java/com/google/neighborgood/IntegrationTest.java

```diff
@@ -0,0 +1,1065 @@
+    assertEquals(recentTask.get("helper"), taskHelper);
+
+    String taskStatus =
+        wait.until(
+                new Function<WebDriver, WebElement>() {
+                  public WebElement apply(WebDriver driver) {
+                    return driver.findElement(By.xpath(awaitVerifTaskXPath + "/td[3]"));
+                  }
+                })
+            .getText();
+    assertEquals(recentTask.get("status"), taskStatus);
+
+    verifyTaskDetails(awaitVerifTaskXPath + "/td[1]");
+
+    // click button to verify a task has been completed
+    wait.until(
+            new Function<WebDriver, WebElement>() {
+              public WebElement apply(WebDriver driver) {
+                return driver.findElement(By.xpath(awaitVerifTaskXPath + "/td[4]/button"));
+              }
+            })
+        .click();
+
+    // Clicking on verify button triggers an alert confirmation window
+    // Driver will try to accept the alert to verify the task every second for a minute
+    for (int i = 0; i < 60; i++) {
+      try {
+        driver.switchTo().alert().accept();
+        break;
+      } catch (NoAlertPresentException e) {
+        driver.manage().timeouts().implicitlyWait(1, TimeUnit.SECONDS);
+      }
+    }
+    // update recent task
+    recentTask.put("status", "COMPLETE");
+    driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
+
+    // Refreshes page due to flakiness of test caused by the partial refresh of the div in the page.
+    // Without refreshing this often resulted in a a stale element error
+    driver.navigate().refresh();
+    driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
+
+    String taskStatusAfter =
+        wait.until(
+                new Function<WebDriver, WebElement>() {
+                  public WebElement apply(WebDriver driver) {
+                    return driver.findElement(By.xpath(awaitVerifTaskXPath + "/td[3]"));
+                  }
+                })
+            .getText();
+    assertEquals(recentTask.get("status"), taskStatusAfter);
+
+    // updates helper's total points
+    helperPoints = Integer.parseInt(recentTask.get("points"));
+    driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
+  }
+
+  @Test
+  /** Verifies that the completed task shows as completed on the helper's user profile */
+  public void _08_Userpage_AsHelper_CompletedTask() {
+    logOut("logout-href");
+    ifLaggingThenRefresh();
+    loginUser(USER_EMAIL_HELPER);
+    ifLaggingThenRefresh();
+    driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
+
+    // Helper's total points
+    String points =
+        wait.until(
+                new Function<WebDriver, WebElement>() {
+                  public WebElement apply(WebDriver driver) {
+                    return driver.findElement(By.id("points"));
+                  }
+                })
+            .getText();
+    assertEquals("My current points: " + Integer.toString(helperPoints) + "pts", points);
```

Comment:
I'm confused about how the value of helperPoints gets set here. Is this test cases definitely executed after the former testcase gets executed?

Reply:
helperPoints get updated on test 7 and all of these tests run in consecutive order :)